core/domain/prod_validation_jobs_one_off.py

```diff
@@ -84,93 +190,2162 @@ def _fetch_external_models(cls, item):
-        for func in cls._get_validation_functions():
+
+        for func in cls._get_custom_validation_functions():
             func(item)
 
 
-class UserSubscriptionsModelValidator(BaseModelValidator):
-    """Class for validating UserSubscriptionsModels."""
+class BaseSummaryModelValidator(BaseModelValidator):
+    """Base class for validating summary models."""
+
+    @classmethod
+    def _get_related_model_properties(cls):
+        """Returns a tuple of related external_models and properties.
+
+        This should be implemented by subclasses.
+
+        Returns:
+            tuple(str, list(tuple), dict): A tuple with first element as
+                related model name, second element as a tuple of
+                cls.external_instance_details and the third element
+                as a properties dict with key as property name in summary
+                model and value as property name in related model.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def _validate_related_model_properties(cls, item):
+        """Validate that properties of the model match the corresponding
+        properties of the related model.
+
+        Args:
+            item: ndb.Model. BaseSummaryModel to validate.
+        """
+
+        for (
+                related_model_name,
+                related_model_class_model_id_model_tuples,
+                related_model_properties_dict
+            ) in cls._get_related_model_properties():
+
+            for (_, _, related_model) in (
+                    related_model_class_model_id_model_tuples):
+                for (property_name, related_model_property_name) in (
+                        related_model_properties_dict.iteritems()):
+                    value_in_summary_model = getattr(item, property_name)
+                    value_in_related_model = getattr(
+                        related_model, related_model_property_name)
+
+                    if value_in_summary_model != value_in_related_model:
+                        cls.errors['%s field check' % property_name].append((
+                            'Entity id %s: %s field in entity: %s does not '
+                            'match corresponding %s %s field: %s') % (
+                                item.id, property_name,
+                                value_in_summary_model,
+                                related_model_name, related_model_property_name,
+                                value_in_related_model))
+
+    @classmethod
+    def validate(cls, item):
+        """Run _fetch_external_instance_details and all _validate functions.
+
+        Args:
+            item: ndb.Model. Entity to validate.
+        """
+        super(BaseSummaryModelValidator, cls).validate(item)
+
+        cls._validate_related_model_properties(item)
+
+
+class BaseSnapshotContentModelValidator(BaseModelValidator):
+    """Base class for validating snapshot content models."""
+
+    model_name = 'snapshot content'
```

Comment:
Add one-line explanatory docstrings for what these are.

Ditto for all class-level constants you introduce.

Reply:
Done.